### DISCUSSION TITLE
Issue with Events Plotting

### DISCUSSION THREAD

**Discussion by angie-john** [pyControl/code#129] [Q&A]
Hi, 

When testing the pyControl box, we ran into an issue where a reward event is plotted when it's not supposed to. What should happen is when a poke occurs, the poke gets plotted and the state changes to the reward state. Then when the mouse goes to get the reward, it should plot the reward with the events only after the mouse receives the reward. However, the issue is that the reward also gets plotted at the same time as the events without a mouse to receive the reward. This issue also happens inconsistently. Sometimes the rewards are plotted, sometimes they aren't. 

We would appreciate any help with figuring out this issue, and we wanted to see if anyone is experiencing something similar. We suspect it has something to do with the hardware or sensors on the box itself.

**Comment by alustig3** [comment]
Could you please share relevant files such as
- task file
- hardware definition file (if you are using one)
- Error log (view->error log). 
 
You can attach a zipped folder with the files, or paste code (see details [here](https://github.com/orgs/pyControl/discussions/65))
